File: Herb/src/main/java/org/Toadflaxmaker/Tasks/Craft.java
package org.Toadflaxmaker.Tasks;

import org.Toadflaxmaker.Task;
import org.powbot.api.Condition;
import org.powbot.api.rt4.*;


public class Craft extends Task {

    public Craft() {
        super();
        this.name = "Crafting";
    }

    public int CraftingWidget = 270;
    public int Potion = 14;

    @Override
    public boolean shouldExecute() {
        return Inventory.stream().name("Toadflax").isNotEmpty()
                && Inventory.stream().name("Vial of water").isNotEmpty();
    }
    @Override
    public void execute() {
        String itemName = "Toadflax";
        String itemName2 = "Vial of water";
        Item Herb = Inventory.stream().name(itemName).first();
        Item Vial_of_water = Inventory.stream().name(itemName2).first();
            if (Herb.valid() && Vial_of_water.valid()) {
                Herb.useOn(Vial_of_water);
                Condition.wait(() -> Widgets.widget(CraftingWidget).component(Potion).visible(), 100, 6);
            }
        if (Widgets.widget(CraftingWidget).component(Potion).visible()) {
            Widgets.widget(CraftingWidget).component(Potion).click();
            Condition.wait(() -> Inventory.stream().name("Toadflax").isEmpty(), 100, 125);
        }
        if (Inventory.stream().name("Toadflax").isEmpty()) {
            if (Inventory.stream().name("Toadflax potion(unf)").count() < 13) {
                System.out.println("Checking to see if this helps with less than 14 pots");
                Bank.open();
            }
        }
    }
}

File: untitled2/src/main/kotlin/crafting/Tasks/Craft.kt
package crafting.Tasks

import crafting.Task
import org.powbot.api.Condition.sleep
import org.powbot.api.Condition.wait
import org.powbot.api.rt4.Inventory.selectedItem
import org.powbot.api.rt4.Inventory.stream
import org.powbot.api.rt4.Widgets.widget
import java.util.concurrent.Callable


class Craft : Task() {
    var CraftingWidget: Int = 270
    var Beer: Int = 14

    init {
        this.name = "Crafting"
    }

    fun shouldExecute(): Boolean {
        return (stream().name("Guam leaf").count() == 14L
                && stream().name("Vial of water").count() == 14L)
    }

    fun execute() {
        val itemName = "Guam leaf"
        val itemName2 = "Vial of water"
        val Guam_leaf = stream().name(itemName).first()
        val Vial_of_water = stream().name(itemName2).first()

        if (selectedItem().id() == -1) {
            Guam_leaf.interact("Use")
            sleep(650)
        } else if (selectedItem().id() == Guam_leaf.id()) {
            Vial_of_water.interact("Use")
            sleep(650)
            if (widget(CraftingWidget).component(Beer).visible()) {
                widget(CraftingWidget).component(Beer).click()
                wait(Callable { stream().name("Guam leaf").isEmpty() }, 85, 100)
            }
        }
    }
}